src/common/prop.c

```diff
@@ -162,44 +163,93 @@ daos_prop_merge(daos_prop_t *old_prop, daos_prop_t *new_prop)
 }
 
 static bool
-daos_prop_str_valid(d_string_t str, const char *prop_name, size_t max_len)
+daos_prop_str_format_valid(const char *str, const char *regex)
+{
+	regex_t		regx;
+	int		rc;
+
+	/* All callers to this internal function shall provide non-NULL args */
+	D_ASSERT(str != NULL);
+	D_ASSERT(regex != NULL);
+
+	rc = regcomp(&regx, regex, REG_EXTENDED | REG_ICASE);
+	if (rc != 0) {
+		D_ERROR("regcomp() failed, %d\n", rc);
+		return false;
+	}
+	rc = regexec(&regx, str, 0, NULL, 0);
+	regfree(&regx);
+	if (rc == 0) {
+		return true;
+	} else if (rc != REG_NOMATCH) {
```

Comment:
The man page seems to say that the return value of regexec can be one of 0 and REG_NOMATCH---no other values possible.